### ISSUE TITLE
Export serializes empty list as list of bools

### ISSUE THREAD

**Issue by SherlockNoMad** [pytorch/pytorch#123480]
### 🐛 Describe the bug

```
class M(torch.nn.Module):
    def forward(self, x):
        return torch.ops.aten.sum.dim_IntList(x, [])

ep = torch.export._trace._export(M(), (x,), strict=False, pre_dispatch=False)

print(torch._export.serde.serialize.serialize(ep).exported_program)
```
shows

"nodes": [{"target": "torch.ops.aten.sum.dim_IntList", "inputs": [{"name": "self", "arg": {"as_tensor": {"name": "arg0_1"}}}, **{"name": "dim", "arg": {"as_bools": []}}**], "outputs": [{"as_tensor": {"name": "sum_1"}}]

This should be serialized as `ints` instead. 

### Versions

main

cc @avikchaudhuri @gmagogsfm @zhxchen17 @tugsbayasgalan @angelayi @suo @ydwu4

**Comment by angelayi** [comment]
lol, I guess it's because we don't actually look at the type of the input [here](https://github.com/pytorch/pytorch/blob/22b99871442fc3c78ec173894c017d6d98996501/torch/_export/serde/serialize.py#L693-L702).

**Comment by SherlockNoMad** [comment]
> lol, I guess it's because we don't actually look at the type of the input [here](https://github.com/pytorch/pytorch/blob/22b99871442fc3c78ec173894c017d6d98996501/torch/_export/serde/serialize.py#L693-L702).

yep, it should read from schema_arg.
